experiments/mock-experiment/bases/base_flower_client.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractmethod
+import flwr as fl
+
+class BaseFlowerClient(fl.client.NumPyClient, ABC):
+    def __init__(self, experiment, fl_model, dataset) -> None:
+        self.experiment = experiment
+        self.fl_model = fl_model
+        self.dataset = dataset
+        super().__init__()
+
+    @abstractmethod
+    def get_parameters(self):
+        """
+        Returns the parameters used for the current model
+        """
+        pass
+
+    @abstractmethod
+    def fit(self, parameters, config):
+        """
+        Fit the model based on the data collected, and return the parameters, length of the training dataset and extra properties
+        """
+        pass
+
+    @abstractmethod
+    def evaluate(self, parameters, config):
```

Comment:
Would you mind explaining your reasoning for not defining a return type for any of the abstract methods?

Reply:
Ali can elaborate for this one, but for some, I think the outputs might vary for some of them.  For example, I found that the evaluation methods are not uniform across different models, like for the linear regression model, we might want to the mean square error.

Reply:
As Laiba mentioned, the outputs are of various types in the child classes, so we refrain from shoehorning any return types to abstract methods. For example, many functions return parameters to send to the server after their training. These parameters are of unknown types, since the parameters for a Logistic Regression Model will differ from the parameters of a Artificial Neural Network, for example. Instead, its easier to let clients define the input and output types for the methods, to fit best with their own use cases.